### PR TITLE
Allowing the use of both objects and arrays for the register()

### PR DESCRIPTION
**Pull request by mivanov93**
Sometimes we may like to use objects for lazy loading and whatsoever and since PHP has the Traversable interface, that should be perfectly fine.

### CODE REVIEW

File: src/Silex/Application.php [hunk 1]
```diff
@@ -162,11 +162,11 @@ public function __construct(array $values = array())
      * Registers a service provider.
      *
      * @param ServiceProviderInterface $provider A ServiceProviderInterface instance
-     * @param array                    $values   An array of values that customizes the provider
+     * @param array|Traversible        $values   An array or a Traversible of values that customizes the provider
```
Comment (mathroc) [inline src/Silex/Application.php]: the name is [`Traversable`](http://php.net/manual/en/class.traversable.php)

File: src/Silex/Application.php [hunk 2]
```diff
@@ -162,11 +162,11 @@ public function __construct(array $values = array())
...
      *
      * @return Application
      */
-    public function register(ServiceProviderInterface $provider, array $values = array())
+    public function register(ServiceProviderInterface $provider, $values = array())
```
Comment (mathroc) [inline src/Silex/Application.php]: instead of just removing `array` you should typehint with `Traversable`

**Comment by mivanov93** [comment]
Thanks, fixed the typo.

**Comment by fabpot** [comment]
Can you rebase this PR (that will remove the merge commit that you have here)?

**Comment by fabpot** [comment]
As the method signature changed, it can only be merged in master.

**Comment by fabpot** [comment]
Thank you @mivanov93.

**Comment by fabpot** [comment]
Reverted as this breaks the Pimple interface. So, we would need to also break BC on Pimple to make this happen, which I think is not worth it.

**Pull request closed without merging**